### PR TITLE
🌱 Release v1.7.0-beta.0

### PR DESCRIPTION
**Pull request by willie-yao**
/area release

### CODE REVIEW

File: CHANGELOG/v1.7.0-beta.0.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
+🚨 This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
+## Highlights
+
+* Enable kubeadm feature gates mutation
+
+## Deprecation Warning
+
+- The API version `v1alpha4` has been completely removed in this release.
```
Comment (fabriziopandini) [inline CHANGELOG/v1.7.0-beta.0.md:9]: Not blocking for the beta release, but I think we should review this statement before GA
cc @sbueringer

**Comment by cahillsf** [comment]
/hold 

this seems valid: https://github.com/kubernetes-sigs/cluster-api/actions/runs/8239613444/job/22533235166?pr=10253

looks like maybe an edge case from the dependencies tool?

File: CHANGELOG/v1.7.0-beta.0.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+- e2e: Test: Fix finalizers test to not only rely on namespaced name (#9891)
+- e2e: Test: retry GetOwnerGraph in owner references test on certificate errors (#10201)
+- IPAM: Fix webhooks using mixed api versions (#9861)
+- KCP: Skip checking `clusterConfiguration.dns` fields when KCP checking MachineNeedRollout (#9857)
+- Logging: Improve log k/v pairs and a improve/drop a few log lines (#9813)
+- Machine: Bubble up machine drain condition in `MachinesReadyCondition` (#9355)
+- Machine: Watch external objects for machine before deleting (#10041)
+- MachinePool: Fix TestReconcileMachinePoolScaleToFromZero flakes (#9745)
+- MULTIPLE_AREAS[API/e2e]: Restore v1alpha3/v1alpha4 conversion to fix SSA issue & add e2e test coverage (#10147)
+- MULTIPLE_AREAS[Testing/CABPK]: Only refresh bootstrap token if needed, requeue in all cases where node hasn't joined yet (#9229)
```
Comment (cahillsf) [inline CHANGELOG/v1.7.0-beta.0.md:73]: these `MULTIPLE_AREAS` can be removed

File: CHANGELOG/v1.7.0-beta.0.md (hunk 1 above)
Comment (sbueringer) [reply CHANGELOG/v1.7.0-beta.0.md:9]: Yup this is not true anymore
